backend/services/user.go

```diff
@@ -1,25 +1,65 @@
 package services
 
 import (
+	"backend/config"
 	"backend/models"
 	"backend/stores"
+
+	"golang.org/x/crypto/bcrypt"
 )
 
 type (
 	UserService interface {
 		GetUserById(id string) (*models.User, error)
-		// CreateUser(user *models.User) (string, error)
-		// UpdateUserById(user *models.User) (string, error)
-		// DeleteUser(id string) error
+		GetUserByUsername(username string) (*models.User, error)
+		CreateUser(creds *config.BasicAuth) (string, error)
+		DeleteUser(id string) error
 	}
 
 	userService struct {
 		stores *stores.Stores
 	}
 )
 
+func (s *userService) CreateUser(creds *config.BasicAuth) (string, error) {
+	encryptedPassword, err := bcrypt.GenerateFromPassword(
+		[]byte(creds.Password),
+		bcrypt.DefaultCost,
+	)
+	if err != nil {
+		return "", err
+	}
+	defaultAdmin := models.Admin{}
+	result := s.stores.DB.First(&defaultAdmin)
+
+	if result.Error != nil {
+		return "", err
+	}
+
+	user := models.User{
+		Username:     creds.Username,
+		PasswordHash: string(encryptedPassword),
+		Admin:        defaultAdmin,
+	}
+	user.PasswordHash = string(encryptedPassword)
+	user.Username = creds.Username
+
+	userId, err := s.stores.User.Create(&user)
+	return userId, err
+}
+
 func (s *userService) GetUserById(id string) (*models.User, error) {
 	var user *models.User
 	user, err := s.stores.User.GetById(id)
 	return user, err
 }
+
+func (s *userService) GetUserByUsername(username string) (*models.User, error) {
+	var user *models.User
+	user, err := s.stores.User.GetByUsername(username)
+	return user, err
```

Comment:
[want]
`:=` で宣言と代入を同時に行なっているので var での宣言は必要なさそうです。
加えて、呼び出したメソッドの帰り値と呼び出し側の帰り値が揃っている場合はそのまま return することができます。
結果的に以下のようになると思います。

```suggestion
	return s.stores.User.GetByUsername(username)
```

Reply:
確かにそうですね！修正しました。